docs/BreakingChanges/081 XmlWriter throws on invalid surrogate pairs.md

```diff
@@ -0,0 +1,52 @@
+## 81: XmlWriter throws on invalid surrogate pairs
```

Comment:
{
  "Id": "81",
  "Title": "XmlWriter throws on invalid surrogate pairs",
  "Details": "For apps that target the .NET Framework 4.5.2 or previous versions, writing an invalid surrogate pair using exception fallback handling does not always throw an exception. For apps that target the .NET Framework 4.6, attempting to write an invalid surrogate pair throws an `ArgumentException`.",
  "Suggestion": "If necessary, this break can be avoided by targeting the .NET Framework 4.5.2 or earlier. Alternatively, invalid surrogate pairs can be pre-processed into valid xml prior to writing them.",
  "Link": "https://msdn.microsoft.com/en-us/library/dn833123(v=vs.110).aspx",
  "Markdown": null,
  "ApplicableApis": [
    "M:System.Xml.XmlWriter.WriteAttributeString(System.String,System.String)",
    "M:System.Xml.XmlWriter.WriteAttributeString(System.String,System.String,System.String)",
    "M:System.Xml.XmlWriter.WriteAttributeString(System.String,System.String,System.String,System.String)",
    "M:System.Xml.XmlWriter.WriteAttributeStringAsync(System.String,System.String,System.String,System.String)",
    "M:System.Xml.XmlWriter.WriteCData(System.String)",
    "M:System.Xml.XmlWriter.WriteCDataAsync(System.String)",
    "M:System.Xml.XmlWriter.WriteChars(System.Char[],System.Int32,System.Int32)",
    "M:System.Xml.XmlWriter.WriteCharsAsync(System.Char[],System.Int32,System.Int32)",
    "M:System.Xml.XmlWriter.WriteComment(System.String)",
    "M:System.Xml.XmlWriter.WriteCommentAsync(System.String)",
    "M:System.Xml.XmlWriter.WriteEntityRef(System.String)",
    "M:System.Xml.XmlWriter.WriteEntityRefAsync(System.String)",
    "M:System.Xml.XmlWriter.WriteRaw(System.Char[],System.Int32,System.Int32)",
    "M:System.Xml.XmlWriter.WriteProcessingInstruction(System.String,System.String)",
    "M:System.Xml.XmlWriter.WriteProcessingInstructionAsync(System.String,System.String)",
    "M:System.Xml.XmlWriter.WriteRaw(System.String)",
    "M:System.Xml.XmlWriter.WriteRawAsync(System.Char[],System.Int32,System.Int32)",
    "M:System.Xml.XmlWriter.WriteRawAsync(System.String)",
    "M:System.Xml.XmlWriter.WriteString(System.String)",
    "M:System.Xml.XmlWriter.WriteStringAsync(System.String)",
    "M:System.Xml.XmlWriter.WriteSurrogateCharEntity(System.Char,System.Char)",
    "M:System.Xml.XmlWriter.WriteSurrogateCharEntityAsync(System.Char,System.Char)",
    "M:System.Xml.XmlWriter.WriteValue(System.String)"
  ],
  "Related": null,
  "VersionBroken": "4.6",
  "VersionFixed": null,
  "IsBuildTime": false,
  "IsQuirked": true,
  "IsSourceAnalyzerAvailable": false,
  "BugLink": null,
  "Notes": "Analyzer wouldn't add much over ApiPort, but we could do basic parameter analysis\nSingle-diagnostic compilation action analyzer?? Depends on how likely it is we think we could usefully analyze arguments to these methods (and whether they contained invalid surrogate pairs)",
  "IsRetargeting": true,
  "ImpactScope": 3
}

**Parsed APIs:**
- [x] M:System.Xml.XmlWriter.WriteAttributeString(System.String,System.String)
- [x] M:System.Xml.XmlWriter.WriteAttributeString(System.String,System.String,System.String)
- [x] M:System.Xml.XmlWriter.WriteAttributeString(System.String,System.String,System.String,System.String)
- [x] M:System.Xml.XmlWriter.WriteAttributeStringAsync(System.String,System.String,System.String,System.String)
- [x] M:System.Xml.XmlWriter.WriteCData(System.String)
- [x] M:System.Xml.XmlWriter.WriteCDataAsync(System.String)
- [x] M:System.Xml.XmlWriter.WriteChars(System.Char[],System.Int32,System.Int32)
- [x] M:System.Xml.XmlWriter.WriteCharsAsync(System.Char[],System.Int32,System.Int32)
- [x] M:System.Xml.XmlWriter.WriteComment(System.String)
- [x] M:System.Xml.XmlWriter.WriteCommentAsync(System.String)
- [x] M:System.Xml.XmlWriter.WriteEntityRef(System.String)
- [x] M:System.Xml.XmlWriter.WriteEntityRefAsync(System.String)
- [x] M:System.Xml.XmlWriter.WriteRaw(System.Char[],System.Int32,System.Int32)
- [x] M:System.Xml.XmlWriter.WriteProcessingInstruction(System.String,System.String)
- [x] M:System.Xml.XmlWriter.WriteProcessingInstructionAsync(System.String,System.String)
- [x] M:System.Xml.XmlWriter.WriteRaw(System.String)
- [x] M:System.Xml.XmlWriter.WriteRawAsync(System.Char[],System.Int32,System.Int32)
- [x] M:System.Xml.XmlWriter.WriteRawAsync(System.String)
- [x] M:System.Xml.XmlWriter.WriteString(System.String)
- [x] M:System.Xml.XmlWriter.WriteStringAsync(System.String)
- [x] M:System.Xml.XmlWriter.WriteSurrogateCharEntity(System.Char,System.Char)
- [x] M:System.Xml.XmlWriter.WriteSurrogateCharEntityAsync(System.Char,System.Char)
- [x] M:System.Xml.XmlWriter.WriteValue(System.String)